apps/andi/lib/andi_web/live/ingestion_live_view/edit_ingestion_live_view.ex

```diff
@@ -8,26 +8,56 @@ defmodule AndiWeb.IngestionLiveView.EditIngestionLiveView do
   alias Andi.Services.IngestionStore
   alias Andi.Services.IngestionDelete
   alias Andi.InputSchemas.InputConverter
+  alias AndiWeb.InputSchemas.IngestionMetadataFormSchema
+  alias Ecto.Changeset
 
   import SmartCity.Event, only: [ingestion_update: 0]
 
   @instance_name Andi.instance_name()
 
   access_levels(render: [:private])
 
+  def mount(
+        _params,
+        %{"is_curator" => is_curator, "ingestion" => ingestion, "user_id" => user_id} = _session,
+        socket
+      ) do
+    default_changeset = InputConverter.andi_ingestion_to_full_ui_changeset(ingestion)
+
+    {:ok,
+      assign(socket,
+        changeset: default_changeset,
+        click_id: nil,
+        delete_ingestion_modal_visibility: "hidden",
+        unsaved_changes_modal_visibility: "hidden",
+        is_curator: is_curator,
+        unsaved_changes: false,
+        page_error: false,
+        ingestion: ingestion,
+        save_success: false,
+        success_message: "",
+        user_id: user_id
+      )}
+  end
+
   def render(assigns) do
+    metadata_changeset = IngestionMetadataFormSchema.extract_from_ingestion_changeset(assigns.changeset)
```

Comment:
Should build all child changesets here and pass down to live_component as assigns.

If the child has UI-only fields that the API/Database wouldn't care about, the child can merge it's current changeset to the changeset provided by the parent on render.

Reply:
If performance becomes an issue (Which I doubt unless changesets are heavier than expected), an optimization may be to detect when the changeset is in the _changed property of the assigns (Meaning it just was changed this render) and only rebuild the changeset then.